executor/src/witgen/generator.rs

```diff
@@ -5,19 +5,16 @@ use std::time::Instant;
 // TODO should use finite field instead of abstract number
 use number::{DegreeType, FieldElement};
 
-use super::affine_expression::AffineExpression;
+use super::affine_expression::{AffineExpression, AffineResult};
 use super::bit_constraints::{BitConstraint, BitConstraintSet};
-use super::eval_error::EvalError;
+
 use super::expression_evaluator::ExpressionEvaluator;
 use super::machines::{FixedLookup, Machine};
 use super::symbolic_witness_evaluator::{SymoblicWitnessEvaluator, WitnessColumnEvaluator};
 use super::util::{contains_next_witness_ref, WitnessColumnNamer};
-use super::{Constraint, EvalResult, FixedData, WitnessColumn};
+use super::{Constraint, EvalResult, EvalValue, FixedData, IncompleteCause, WitnessColumn};
 
-pub struct Generator<'a, QueryCallback>
-where
-    QueryCallback: FnMut(&'a str) -> Option<FieldElement>,
-{
+pub struct Generator<'a, QueryCallback> {
```

Comment:
It's discouraged to put trait bounds on struct declarations unless necessary. They belong on implementations instead.